server/src/main/java/io/druid/server/QueryLifecycle.java

```diff
@@ -183,41 +186,68 @@ public void initialize(final Query baseQuery)
   /**
    * Authorize the query. Will return an Access object denoting whether the query is authorized or not.
    *
-   * @param authorizationInfo authorization info from the request; or null if none is present. This must be non-null
-   *                          if security is enabled, or the request will be considered unauthorized.
+   * @param token authentication token from the request
+   * @param namespace namespace of the authentication token
+   * @param authenticationResult authentication result of the request
    *
    * @return authorization result
    *
-   * @throws IllegalStateException if security is enabled and authorizationInfo is null
-   */
-  public Access authorize(@Nullable final AuthorizationInfo authorizationInfo)
+   * */
+  public Access authorize(
+      @Nullable final AuthenticationResult authenticationResult
```

Comment:
It seems like authenticationResult is not actually nullable, so remove the annotation.